### PR TITLE
io: add io copy function

### PR DESCRIPTION
**Pull request by LouisSchmieder**
This PR add's a io copy function `cp` which copies a reader src directly to a writer destination

### CODE REVIEW

File: vlib/io/io.v [hunk 1]
```diff
@@ -0,0 +1,17 @@
+module io
+
+const (
+	buf_max_len = 5 * 1024
```
Comment (spytheman) [inline vlib/io/io.v:4]: why is that needed?

File: vlib/io/io.v [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+	mut buf := read_all(reader: src) or {
+		return err
+	}
+	dst.write(buf) or {
+		return
+	}
+	unsafe {
+		buf.free()
+	}
+}
```
Comment (spytheman) [inline vlib/io/io.v:17]: That would read everything from src in memory, which is fine for a first draft, but it will be better to copy chunks from src to dst as they come.

Comment (JalonSolov) [reply vlib/io/io.v:17]: That would be one reason to have the const listed above...

Comment (spytheman) [reply vlib/io/io.v:17]: yes, but it is not used as it is - perhaps it would be in the future?

Comment (LouisSchmieder) [reply vlib/io/io.v:17]: I will add it in future, because that's just a base for now
